File: src/main/java/top/gerritchang/tools/DemoApplication.java
package top.gerritchang.tools;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.ComponentScan;
import top.gerritchang.check.websocket.CommonSocket;

@SpringBootApplication
@ComponentScan({"top.gerritchang.check","top.gerritchang.tools"})
public class DemoApplication {
    public static void main(String[] args) {
        SpringApplication springApplication = new SpringApplication(DemoApplication.class);
        ConfigurableApplicationContext configurableApplicationContext = springApplication.run(args);
        CommonSocket socket = new CommonSocket();
        socket.setApplicationContext(configurableApplicationContext);
    }
}

File: src/main/java/top/gerritchang/tools/DemoApplication.kt
package top.gerritchang.tools

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.context.annotation.ComponentScan
import top.gerritchang.check.websocket.CommonSocket

@SpringBootApplication
@ComponentScan("top.gerritchang.check","top.gerritchang.tools")
open class DemoApplication

fun main(args: Array<String>) {
    val springApplication = SpringApplication(DemoApplication::class.java)
    val configurableApplicationContext: ConfigurableApplicationContext = springApplication.run(*args)
    val socket = CommonSocket()
    socket.setApplicationContext(configurableApplicationContext)
}
